### PR TITLE
Provisioning via DBus

### PR DESCRIPTION
**Pull request by adamgralec-silvair**

### CODE REVIEW

File: mesh/mesh-db.c [hunk 1]
```diff
@@ -1250,16 +1250,16 @@ bool mesh_db_read_node(json_object *jnode, mesh_db_node_cb cb, void *user_data)
...
-	if (!parse_iv_idx(jnode, &node))
-		l_info("Failed to parse IV index and IV update");
-
 	/* Parse UUID */
 	if (!parse_uuid(jnode, &node)) {
 		l_info("Failed to parse uuid");
 		return false;
 	}
 
+	/* Parse IV idx and IV update flag */
```
Comment (michallowasrzechonek-silvair) [inline mesh/mesh-db.c]: Any reason why this was moved after UUID?

File: mesh/mesh.c [hunk 2]
```diff
@@ -110,6 +110,11 @@ static void start_io(uint16_t index)
 	node_attach_io(io);
 }
 
+struct mesh_io *get_mesh_io(void)
```
Comment (michallowasrzechonek-silvair) [inline mesh/mesh.c]: Rename to `mesh_get_io`, functions in the module should have a common prefix.

File: mesh/node.c [hunk 3]
```diff
@@ -1265,16 +1274,45 @@ bool delete_node(uint8_t *uuid)
...
-	return true;
+	struct mesh_prov_node_info *info;
+	char *str;
+	bool status = false;
+
+	if (node->net)
+		return false;
+
+	str = l_util_hexstring(network_key, KEY_LEN);
+	l_debug("Provisioning node with netkey: %s", str);
```
Comment (michallowasrzechonek-silvair) [inline mesh/node.c]: Remove debug and `str` local var.

File: mesh/node.c [hunk 4]
```diff
@@ -1776,27 +1830,26 @@ bool node_add_pending_local(struct mesh_node *node, void *prov_node_info,
...
-
-	l_queue_push_tail(nodes, node);
-
+	/* Update IV index and IV update in Json file */
 	if (!storage_set_iv_index(node->net, info->iv_index, ivu))
 		return false;
 
 	mesh_net_set_iv_index(node->net, info->iv_index, ivu);
 
+    /* Update unicast address in Json file */
```
Comment (michallowasrzechonek-silvair) [inline mesh/node.c]: Indent

File: mesh/node.c [hunk 5]
```diff
@@ -1776,27 +1830,26 @@ bool node_add_pending_local(struct mesh_node *node, void *prov_node_info,
...
 	if (!mesh_db_write_uint16_hex(node->jconfig, "unicastAddress",
 								info->unicast))
 		return false;
 
 	node->primary = info->unicast;
 	mesh_net_register_unicast(node->net, info->unicast, node->num_ele);
 
-	memcpy(node->dev_key, info->device_key, 16);
+    /* Update device key in Json file */
```
Comment (michallowasrzechonek-silvair) [inline mesh/node.c]: Indent

File: mesh/node.c [hunk 6]
```diff
@@ -1776,27 +1830,26 @@ bool node_add_pending_local(struct mesh_node *node, void *prov_node_info,
...
+	memcpy(node->dev_key, info->device_key, KEY_LEN);
 	if (!mesh_db_write_device_key(node->jconfig, info->device_key))
 		return false;
 
+    /* Update network key parameters in Json file */
```
Comment (michallowasrzechonek-silvair) [inline mesh/node.c]: Indent

File: mesh/storage.c [hunk 7]
```diff
@@ -92,8 +92,10 @@ static bool read_net_keys_cb(uint16_t idx, uint8_t *key, uint8_t *new_key,
 	if (!net)
 		return false;
 
-	if (mesh_net_add_key(net, false, idx, key) != MESH_STATUS_SUCCESS)
+	if (mesh_net_add_key(net, false, idx, key) != MESH_STATUS_SUCCESS) {
+		l_debug("cannot add net key");
```
Comment (michallowasrzechonek-silvair) [inline mesh/storage.c]: I'd say this is an error (or at least a warning), not debug.

File: mesh/storage.c [hunk 8]
```diff
@@ -446,6 +466,7 @@ bool storage_load_nodes(const char *dir_name)
 
 	while ((entry = readdir(dir)) != NULL) {
 		char name_buf[PATH_MAX];
+		char name_with_uuid_buf[PATH_MAX + UUID_LEN];
```
Comment (michallowasrzechonek-silvair) [inline mesh/storage.c]: This doesn't seem used anywhere. Besides, paths longer than `PATH_MAX` won't work anyway. See `<linux/limits.h>`

**Review by michallowasrzechonek-silvair — CHANGES_REQUESTED** [review]

**Comment by michallowasrzechonek-silvair** [comment]
Other than a few minor comments, LGTM. Nice work :+1:

File: mesh/storage.c (hunk 7 above)
Comment (jakubwitowski-silvair) [reply mesh/storage.c]: Done

File: mesh/storage.c (hunk 8 above)
Comment (jakubwitowski-silvair) [reply mesh/storage.c]: It is used to delete ..some_path/node_uuid_nr directory when cannot parse neither json nor json.bak files.

Size fixed

File: mesh/mesh-db.c (hunk 1 above)
Comment (jakubwitowski-silvair) [reply mesh/mesh-db.c]: No reason - fixed

File: mesh/storage.c (hunk 8 above)
Comment (michallowasrzechonek-silvair) [reply mesh/storage.c]: https://linux.die.net/man/3/dirname

File: mesh/mesh.c (hunk 2 above)
Comment (adamgralec-silvair) [reply mesh/mesh.c]: fixed

File: mesh/node.c (hunk 3 above)
Comment (adamgralec-silvair) [reply mesh/node.c]: fixed

File: mesh/node.c (hunk 4 above)
Comment (adamgralec-silvair) [reply mesh/node.c]: fixed

File: mesh/node.c (hunk 5 above)
Comment (adamgralec-silvair) [reply mesh/node.c]: fixed

File: mesh/node.c (hunk 6 above)
Comment (adamgralec-silvair) [reply mesh/node.c]: fixed

File: mesh/storage.c (hunk 8 above)
Comment (jakubwitowski-silvair) [reply mesh/storage.c]: Done

**Review by michallowasrzechonek-silvair — APPROVED** [review]

**Pull request merged**